src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Behaviors/ListSearchPipeBehavior.cs

```diff
@@ -0,0 +1,131 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Behavior
+{
+    /// <summary>
+    /// This behavior looks for searches that use _list then resolves the list and rewrites the query with the resource _ids
+    /// </summary>
+    public class ListSearchPipeBehavior : IPipelineBehavior<SearchResourceRequest, SearchResourceResponse>
+    {
+        private const string _listParameter = "_list";
+        private const string _idParameter = "_id";
+        private readonly IScoped<IFhirDataStore> _dataStore;
+        private readonly ResourceDeserializer _deserializer;
+        private readonly IBundleFactory _bundleFactory;
+        private readonly IReferenceSearchValueParser _referenceSearchValueParser;
+        private readonly ISearchOptionsFactory _searchOptionsFactory;
+
+        public ListSearchPipeBehavior(
+            ISearchOptionsFactory searchOptionsFactory,
+            IBundleFactory bundleFactory,
+            IScoped<IFhirDataStore> dataStore,
+            ResourceDeserializer deserializer,
+            IReferenceSearchValueParser referenceSearchValueParser)
+        {
+            EnsureArg.IsNotNull(bundleFactory, nameof(bundleFactory));
+            EnsureArg.IsNotNull(dataStore, nameof(dataStore));
+            EnsureArg.IsNotNull(deserializer, nameof(deserializer));
+            EnsureArg.IsNotNull(referenceSearchValueParser, nameof(referenceSearchValueParser));
+            EnsureArg.IsNotNull(searchOptionsFactory, nameof(searchOptionsFactory));
+
+            _searchOptionsFactory = searchOptionsFactory;
+            _bundleFactory = bundleFactory;
+            _dataStore = dataStore;
+            _deserializer = deserializer;
+            _referenceSearchValueParser = referenceSearchValueParser;
+        }
+
+        public async Task<SearchResourceResponse> Handle(SearchResourceRequest request, CancellationToken cancellationToken, RequestHandlerDelegate<SearchResourceResponse> next)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            Tuple<string, string> listParameter = request.Queries
+                .FirstOrDefault(x => string.Equals(x.Item1, _listParameter, StringComparison.Ordinal));
+
+            // if _list was not requested, continue...
+            if (listParameter == null)
+            {
+                return await next();
+            }
+
+            // create an empty response to return if needed
+            SearchResourceResponse emptyResponse = CreateEmptySearchResponse(request);
```

Comment:
I realize this is just for code cleanliness, but is it really required? 
I'm asking since the method creating it isn't that simple so creating the object each time even if not used might be too much... IMO using `return CreateEmptySearchResponse(request)` each time isn't that much of a duplication of code.

Reply:
im good with both options. I will change as it will add (minor) perf upgrade